prow/cmd/checkconfig/main.go

```diff
@@ -473,3 +481,17 @@ func validateNeedsOkToTestLabel(cfg *config.Config) error {
 	}
 	return errorutil.NewAggregate(queryErrors...)
 }
+
+func validateBranchProtectionConfig(cfg *config.Config) error {
+	var configErrors []error
+	for org := range cfg.BranchProtection.Orgs {
+		for repo := range cfg.BranchProtection.Orgs[org].Repos {
+			for branch := range cfg.BranchProtection.Orgs[org].Repos[repo].Branches {
+				if _, err := cfg.GetBranchProtection(org, repo, branch); err != nil {
+					configErrors = append(configErrors, err)
```

Comment:
This seems like it is detecting config errors, not potential misconfigurations that aren't actually errors. Shouldn't it belong in the config parsing logic instead of the `checkconfig` binary?

Reply:
GetBranchProtection is a little more complicated than that, because it merges all the parent declarations into the child ones, returning an error if anything conflicting has happened (aka parent requires cla/whatever context but some branch sets protection: false


Ideally this would note the `org/repo=branch` that has the error (or maybe that's already in the error)?

Reply:
> Shouldn't it belong in the config parsing logic instead of the checkconfig binary?

It also does, but since the list of orgs and branches is dynamic, it's called on an as-needed basis in `branchprotector`

Reply:
> Ideally this would note the org/repo=branch that has the error (or maybe that's already in the error)?

Your errors already contain this information

Reply:
> It also does, but since the list of orgs and branches is dynamic, it's called on an as-needed basis in branchprotector

Not sure I follow. Can't we just move this `validateBranchProtectionConfig` function into the config parsing logic?